File: OnlinePK-Android/biz-live/src/main/java/com/netease/biz_live/yunxin/live/model/SeatMemberInfo.java
/*
 * Copyright (c) 2021 NetEase, Inc.  All rights reserved.
 * Use of this source code is governed by a MIT license that can be found in the LICENSE file.
 */

package com.netease.biz_live.yunxin.live.model;

import java.io.Serializable;
import java.util.Objects;

public class SeatMemberInfo implements Serializable {
    public String accountId;
    public String nickName;
    public String avatar;
    public int audio;
    public int video;
    public String avRoomCName;
    public String avRoomCid;
    public long avRoomUid;
    public String avRoomCheckSum;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SeatMemberInfo that = (SeatMemberInfo) o;
        return Objects.equals(accountId, that.accountId);
    }

    @Override
    public int hashCode() {
        return Objects.hash(accountId);
    }

}

File: OnlinePK-Android/biz-live/src/main/java/com/netease/biz_live/yunxin/live/model/SeatMemberInfo.kt
/*
 *  Copyright (c) 2021 NetEase, Inc.  All rights reserved.
 *  Use of this source code is governed by a MIT license that can be found in the LICENSE file
 */

package com.netease.biz_live.yunxin.live.model

import com.netease.yunxin.seatlibrary.seat.event.AvRoomUser
import com.netease.yunxin.seatlibrary.seat.model.SeatInfo
import java.io.Serializable
import java.util.*

class SeatMemberInfo(
    /**
     * 麦位信息
     */
    val seatInfo: SeatInfo,
    /**
     * 用户rtc信息
     */
    var avRoomUser: AvRoomUser? = null,

    /**
     * 是否是自己
     */
    var isSelf: Boolean = false
) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as SeatMemberInfo?
        return seatInfo.accountId == that?.seatInfo?.accountId
    }

    override fun hashCode(): Int {
        return Objects.hash(seatInfo.accountId)
    }
}